service/src/test/java/uk/nhs/adaptors/gp2gp/tasks/TaskDefinitionFactoryTest.java

```diff
@@ -0,0 +1,48 @@
+package uk.nhs.adaptors.gp2gp.tasks;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import static uk.nhs.adaptors.gp2gp.common.constants.Constants.DOCUMENT_TASK;
+import static uk.nhs.adaptors.gp2gp.common.constants.Constants.STRUCTURE_TASK;
+
+import java.util.Optional;
+import java.util.stream.Stream;
+
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+
+@SpringBootTest
```

Comment:
SpringBootTest should not be used for unit test. This starts a spring boot application for every test!

Reply:
Will need some modifications to make this a unit test rather than a component test